cmake/Modules/FindVSX.cmake

```diff
@@ -0,0 +1,45 @@
+
+IF(CMAKE_SYSTEM_NAME MATCHES "Linux")
+   message("<FindVSX>")
+
+   EXEC_PROGRAM(LD_SHOW_AUXV=1 ARGS "/bin/true" OUTPUT_VARIABLE bintrue) 
+   if(bintrue MATCHES "AT_PLATFORM:[ \\t\\n\\r]*([a-zA-Z0-9_]+)[ \\t\\n\\r]*")
+    if(CMAKE_MATCH_COUNT GREATER 0)
+     string(TOLOWER ${CMAKE_MATCH_1} platform)
+     if(${platform} MATCHES "^power")
+     message("POWER Platform: ${platform}")
+         SET(POWER_COMP TRUE CACHE BOOL "power ")
+         SET(CXX_VSX_FLAGS  "${CXX_VSX_FLAGS} -mcpu=${platform} -mtune=${platform}" )
+     endif()
+   endif()
+   endif()
+   if(POWER_COMP AND bintrue MATCHES "AT_HWCAP:.*(vsx).*")
```

Comment:
What's the minimum CPU version for Pytorch on Power? If it's POWER7 and above, you can assume VSX is present.

Reply:
@ceseo yes right. I remember once because of virtual hypervisor vector instructions were prohibited for z13. Probably that's not case for power

Reply:
Thanks for the technical review @ceseo !

The issue is not missing instructions, but compiler earlier than GCC 8 don't define the vec_float type.  Maybe it's easiest, as Carlos suggested, to conditionally define the type for compilers earlier than GCC 8.

```
#define vec_float __builtin_vec_float
```

Reply:
yes, I already mentioned I will add asm inline definition for it in the header

Reply:
I'll  keep platform name check and compiler vsx check ,here